### PR TITLE
Structuring Large Applications with Blueprint and Application Factory…

### PR DESCRIPTION
**Pull request by corpsgeek**
… in Flask

### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by corpsgeek** [comment]
This pull request fixes issue https://github.com/section-engineering-education/engineering-education/issues/2365

**Comment by ahmadmardeni1** [comment]
👋

**Comment by hectorkambow** [comment]
@corpsgeek 
Good afternoon and thank you for submitting your article for publishing. It has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it.  🚀

**Comment by corpsgeek** [comment]
Alright @hectorkambow

**Comment by hectorkambow** [comment]
plagiarism check done

File: articles/structuring-large-applications-with-blueprints-and-application-factory-in-flask/index.md [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+- [Working with Configuration and Environment Variable](#working-with-configuration-and-environmental-variables)
+- [Structuring Applications with Blueprint and Application Factory](#structuring-a-large-scale-application)
+- [Conclusion](#conclusion)
+
+### Work Environment Setup
+In this tutorial, the use of PyCharm Integrated Development Environment is recommended, because it has all the features required to set up a basic application. 
+
+However, if you have a different preference in terms of IDE, then you should consider installing a virtual environment and installing Flask.
+
+I intend to not float the introductory section of this article with guides on how to install a Flask web application and setting up your work environment. If you have no fundamental knowledge on setting up your work environment to develop a basic Flask application, then read my introductory article on the Flask framework [here](https://www.section.io/engineering-education/complete-guide-on-installing-flask-for-beginners/).
```
Comment (srishilesh) [inline articles/structuring-large-applications-with-blueprints-and-application-factory-in-flask/index.md:30]:
```suggestion
I intend to not float the introductory section of this article with guides on how to install a Flask web application and setting up your work environment. If you have no fundamental knowledge on setting up your work environment to develop a basic Flask application, then read my introductory article on the Flask framework [here](/engineering-education/complete-guide-on-installing-flask-for-beginners/).
```

**Review by srishilesh — CHANGES_REQUESTED** [review]
Hey @corpsgeek,

Great article.

The preliminary revision is done. Before we proceed to tech review, please use Grammarly to perform a grammar check.

Let me know when it's done.

Thanks.

File: articles/structuring-large-applications-with-blueprints-and-application-factory-in-flask/index.md (hunk 1 above)
Comment (corpsgeek) [reply articles/structuring-large-applications-with-blueprints-and-application-factory-in-flask/index.md]: Resolved.

**Comment by corpsgeek** [comment]
Done. @srishilesh

**Review by srishilesh — APPROVED** [review]
Hey @corpsgeek,

Amazing article. It was very beginner friendly.

Moving for the final review.

Thanks.

**Comment by louisefindlay23** [comment]
Hi, @corpsgeek. 

You should use a separate branch on your forked repo for article submissions, not the master branch. Please create a new branch from master so we can conduct a final review of your article.

cc @srishilesh

**Comment by corpsgeek** [comment]
@srishilesh how do I go about fixing the above issue?

**Comment by corpsgeek** [comment]
@louisefindlay23 here is the PR in a new branch and all changes https://github.com/section-engineering-education/engineering-education/pull/2905

**Pull request closed without merging**